jgrapht-core/src/main/java/org/jgrapht/alg/KDisjointShortestPaths.java

```diff
@@ -0,0 +1,387 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2016, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* -------------------------
+ * .java
+ * -------------------------
+ * (C) Copyright 2007-2016, by France Telecom
+ *
+ * Original Author: Assaf Mizrachi and Contributors.
+ * Contributor(s):
+ *
+ * $Id$
+ *
+ * Changes
+ * -------
+ * 11-Sep-2016 : Initial revision (AM);
+ *
+ */
+package org.jgrapht.alg;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.WeightedGraph;
+import org.jgrapht.graph.AsWeightedDirectedGraph;
+
+/**
+ * The algorithm determines the k <em>disjoint</em> shortest simple paths in increasing order of
+ * weight. Weights can be negative (but no negative cycle is allowed). Only directed simple graphs
+ * are allowed. 
+ * <br>
+ * An undirected graph can be transformed to a directed simple graph using the following
+ * transformation:
+ * <br>
+ * For each pair of vertices V, U and an undirected edge E:V<-->U with weight W:
+ * <ol>
+ * <li> Create vertices V_in, V_out, U_in and U_out
+ * <li> Create edges:
+ * 	<ol>
+ * 	<li> V_out-->U_in, weight = W;
+ *  <li> U_out-->V_in, weight = W;
+ *  <li> V_in-->V_out, weight = 0;
+ *  <li> U_in-->U_out, weight = 0;
+ *  </ol>
+ * </ol>
+ * <p>
+ * The algorithm is based on the Surballe (later extended by Bhandari) algorithm, so is to find an
+ * Edge-disjoint shortest paths. In order to find a Vertex-disjoint shortest paths use the

```

Comment:
This is the main one: https://www.nas.ewi.tudelft.nl/people/Fernando/papers/Wiley.pdf